packages/vite/src/node/plugins/json.ts

```diff
@@ -48,7 +48,7 @@ export function jsonPlugin(
       json = stripBomTag(json)
 
       try {
-        if (options.stringify !== false) {
+        if (options.stringify !== false && /^\s*\{/.test(json)) {
           if (options.namedExports) {
             const parsed = JSON.parse(json)
             if (typeof parsed === 'object' && parsed != null) {
```

Comment:
```suggestion
        if (options.stringify !== false) {
          if (options.namedExports && /^\s*\{/.test(json)) {
            const parsed = JSON.parse(json)
```
(additionally `}` at line 74 needs to be removed)

I think the object check should be after `options.stringify !== false`. Arrays should be stringified if `stringify` is true or auto.

Reply:
I made the change and I didn't remove the line 74 because that would break the file since no open curly was removed and so there's no ending curly to remove either. I also added the extra test you suggested.